scripts/pages/home/model.mjs

```diff
@@ -0,0 +1,236 @@
+import {
+    PLAYER_ID,
+    GET_HEROES_ID_URL,
+    GET_PROFILE_INFO_URL,
+    GET_PROFILE_WL_URL,
+    GET_MATCHES_HISTORY_URL,
+    GET_MATCH_DETAILS_URL,
+    STEAM_API_REQUEST_PARAMS,
+    OPENDOTA_REQUEST_OPTIONS,
+    RAPID_API_REQUEST_OPTIONS,
+    RAPID_API_MATCHES_URL_PARAMS,
+    STEAM_API_HEROES_URL_PARAMS,
+    buildUrlWithParams,
+} from "../../api/api.mjs"
+
+import {
+    DataManager
+} from "../../utils/dataManager.mjs"
+
+export default {
+
+    async getProfileInfo() {
+        let profileInfoData = null;
+        let profileWlData = null;
+
+        const profileInfoDataManager = new DataManager(GET_PROFILE_INFO_URL, { requestOptions: OPENDOTA_REQUEST_OPTIONS });
+
+        if (profileInfoDataManager.cache) {
+            profileInfoData = profileInfoDataManager.cache;
+        } else {
+            profileInfoData = await profileInfoDataManager.fetch();
+        }
+
+        const profileWlDataManager = new DataManager(GET_PROFILE_WL_URL, { requestOptions: OPENDOTA_REQUEST_OPTIONS });
+
+        if (profileWlDataManager.cache) {
+            profileWlData = profileWlDataManager.cache;
+        } else {
+            profileWlData = await profileWlDataManager.fetch();
+        }
+
+        let profileInfo = {
+            personaname: profileInfoData.profile.personaname,
+            avatarfull: profileInfoData.profile.avatarfull,
+            wl: profileWlData,
+            winrate: (profileWlData.win / (profileWlData.win + profileWlData.lose) * 100).toFixed(2),
+        };
+
+        return profileInfo;
+    },
+
+    async getMatchesInfo() {
+        let matchesInfoData = [];
+
+        let heroesIdData = null;
+        let itemsIdData = null;
+
+        $.getJSON('json/heroes-id.json', function (result) {
+            heroesIdData = result.result.heroes;
+            //console.log(heroesIdData);
+        })
+
+        $.getJSON('json/items-id.json', function (result) {
+            itemsIdData = result;
+            console.log(itemsIdData);
+        })
+
+        const matchesDataManager = new DataManager(GET_MATCHES_HISTORY_URL, { params: RAPID_API_MATCHES_URL_PARAMS, requestOptions: RAPID_API_REQUEST_OPTIONS });
+
+        const matchesData = await matchesDataManager.fetch();
+        const matches = matchesData.result.matches;
+
+        console.log(matches);
+
+        if (matches.length === 0) {
+            throw new Error("No matches!");
+        }
+
+        const callbackQueue = [];
```

Comment:
For an outer developer, this variable name does not tell much. And in this case, naming is not enough. I would leave a comment like this:
`// The array below consists of fetching callbacks to be executed in order with a delay to prevent the API provider from blocking them`
`const fetchQueue = []`